matRad_StfVMATPost.m

```diff
@@ -0,0 +1,311 @@
+function stf = matRad_StfVMATPost(stf,pln,masterRayPosBEV,masterTargetPointBEV,SAD,machine)
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% matRad steering information post-processing for VMAT
+%
+% call
+%   stf = matRad_StfVMATPost(stf,pln)
+%
+% input
+%   stf:        matRad steering information struct
+%   pln:        matRad plan meta information struct
+%
+% output
+%   stf:        matRad steering information struct
+%
+% References
+%   -
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2015 the matRad development team.
+%
+% This file is part of the matRad project. It is subject to the license
+% terms in the LICENSE file found in the top-level directory of this
+% distribution and at https://github.com/e0404/matRad/LICENSES.txt. No part
+% of the matRad project, including this file, may be copied, modified,
+% propagated, or distributed except according to the terms contained in the
+% LICENSE file.
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%After all steering file information is completed, loop over
+%initialized gantry angles.  All children and subchildren of these angles should
+%have ray positions given by the union of their own ray positions and
+%the ray positions of the parent transformed to their gantry angle.
+%This is so that: (1) the target is still totally in the FOV of each
+%angle; and (2) the parent can give segments to the children during
+%initial segmentation and DAO.
+
+fprintf('matRad: VMAT post-processing (1/2)... ');
+
+numDAO = 1;
+DAODoseAngleBorders = zeros(2*numel(pln.propStf.DAOGantryAngles),1);
+offset = 1;
+timeFacIndOffset = 1;
+
+for i = 1:length(pln.propStf.gantryAngles)
+    
+    % Determine which FMO beam the current beam belongs to
+    [~,stf(i).propVMAT.beamParentFMOIndex] = min(abs(pln.propStf.FMOGantryAngles-pln.propStf.gantryAngles(i)));
+    stf(i).propVMAT.beamParentGantryAngle = pln.propStf.FMOGantryAngles(stf(i).propVMAT.beamParentFMOIndex);
+    stf(i).propVMAT.beamParentIndex = find(abs(pln.propStf.gantryAngles - stf(i).propVMAT.beamParentGantryAngle) < 1e-6);
```

Comment:
For some reason this can set an empty beamParentIndex for some gantry angle spacings. This then will fail later in the function when non-empty beamParentIndex is required. Could this also be a min(abs(...)) operation?

Reply:
My settings:
`
pln.propOpt.VMAToptions.maxGantryAngleSpacing    = 8;      % Max gantry angle spacing for dose calculation
pln.propOpt.VMAToptions.maxDAOGantryAngleSpacing = 16;      % Max gantry angle spacing for DAO
pln.propOpt.VMAToptions.maxFMOGantryAngleSpacing = 112;     % Max gantry angle spacing for FMO

pln.propOpt.VMAToptions.startingAngle = 0;
pln.propOpt.VMAToptions.finishingAngle = 359;
pln.propOpt.VMAToptions.continuousAperture = 0;
`

Reply:
One more remark: Such find operations exist multiple times and cause problems with certain angle and spacing combinations. Do I do something wrong or stupid? Nevertheless, nonsensical parameter combinations should be catched somewhere